airbyte-lib/pyproject.toml

```diff
@@ -1,29 +1,146 @@
+[build-system]
+requires = ["poetry-core"]
+build-backend = "poetry.core.masonry.api"
+
 [tool.poetry]
 name = "airbyte-lib"
+description = "AirbyteLib"
 version = "0.1.0"
-description = ""
-authors = ["Joe Reuter <joe@airbyte.io>"]
+authors = ["Airbyte <contact@airbyte.io>"]
 readme = "README.md"
+packages = [{include = "airbyte_lib"}]
 
 [tool.poetry.dependencies]
 python = "^3.10"
+
+airbyte-cdk = "^0.57.6"
 jsonschema = "3.2.0"
+overrides = "^7.4.0"
+pandas = "^2.1.4"
+pyarrow = "^14.0.1"
+python-ulid = "^2.2.0"
 requests = "^2.31.0"
-airbyte-protocol-models = "^1.0.1"
+SQLAlchemy = "^2.0.23"
+orjson = "^3.9.10"
+duckdb-engine = "^0.9.5"
+psycopg = {extras = ["binary", "pool"], version = "^3.1.16"}
+ulid = "^1.1"
 
 [tool.poetry.group.dev.dependencies]
-pytest = "^7.4.3"
+faker = "^21.0.0"
 mypy = "^1.7.1"
-types-requests = "^2.31.0.10"
-types-jsonschema = "^4.20.0.0"
+pandas-stubs = "^2.1.4.231218"
+pyarrow-stubs = "^10.0.1.7"
+pytest = "^7.4.3"
 pytest-mypy = "^0.10.3"
+ruff = "^0.1.7"
+types-jsonschema = "^4.20.0.0"
+types-requests = "^2.31.0.10"
+pytest-postgresql = "^5.0.0"
+docker = "^7.0.0"
+pytest-docker = "^2.0.1"
 
-[build-system]
-requires = ["poetry-core"]
-build-backend = "poetry.core.masonry.api"
+[tool.pytest.ini_options]
+# addopts = "--mypy"  # DEBUG: This blocks test discovery and execution, so I'm disabling it for now
```

Comment:
Sure thing.

In the VS Code IDE, test discovery was failing when there were mypy issues, causing the test interface to essentially disappear - and you have to go digging in a different output window to find the mypy issue. I can debug a bit more, but that's the symptom I was dealing with. Since it was slowing me down, I disabled for my work. I can try enabling again and see if there are ways around it.

Reply:
The thing that's important to me is that CI is checking mypy for every PR - if we can find another way to ensure that would be fine for me as well.

Currently, airbyte-lib is tested via airbyte-ci which just executes pytest, that's what makes this integration attractive.